src/condition/cmdi_detector.cpp

```diff
@@ -382,20 +375,20 @@ std::optional<shi_result> cmdi_impl(const ddwaf_object &exec_args,
         return std::nullopt;
     }
 
-    object::kv_iterator it(&params, {}, objects_excluded, limits);
+    kv_iterator it(params, {}, objects_excluded, limits);
     for (; it; ++it) {
         if (deadline.expired()) {
             throw ddwaf::timeout_exception();
         }
 
-        const ddwaf_object &param = *(*it);
-        if (param.type != DDWAF_OBJ_STRING) {
+        const object_view param = *it;
+        if (!param.is<std::string_view>()) {
```

Comment:
Why not just is_string, is_unsigned, is_signed, is_bool? I mean, this doesn't read right. param is is not a string_view